src/baldr/edgeinfo.cc

```diff
@@ -445,6 +456,39 @@ json::MapPtr EdgeInfo::json() const {
     edge_info->emplace("speed_limit", static_cast<uint64_t>(speed_limit()));
   }
 
+  json::MapPtr conditional_speed_limits;
+  for (const auto& [tag, value] : GetTags()) {
+    switch (tag) {
+      case TaggedValue::kLayer:
+        break;
+      case TaggedValue::kLinguistic:
+        break;
+      case TaggedValue::kBssInfo:
+        break;
+      case TaggedValue::kLevel:
+        break;
+      case TaggedValue::kLevelRef:
+        break;
+      case TaggedValue::kLandmark:
+        break;
+      case TaggedValue::kConditionalSpeedLimits: {
+        if (!conditional_speed_limits) {
+          conditional_speed_limits = json::map({});
+        }
+        const ConditionalSpeedLimit* l = reinterpret_cast<const ConditionalSpeedLimit*>(value.data());
+        conditional_speed_limits->emplace(l->td_.to_string(), static_cast<uint64_t>(l->speed_));
+        break;
+      }
+      case TaggedValue::kTunnel:
+        break;
+      case TaggedValue::kBridge:
+        break;
+    }
+  }
+  if (conditional_speed_limits) {
+    edge_info->emplace("conditional_speed_limits", std::move(conditional_speed_limits));
+  }
```

Comment:
Well, probably it would be more aligned with other parts to always send `"conditional_speed_limits"` even if empty, but the idea of adding quite a big string to the response where it would be rarely used hurts me.

Reply:
i prefer the way you did it!